crates/masking/src/cassandra.rs

```diff
@@ -0,0 +1,50 @@
+use scylla::{
+    cql_to_rust::FromCqlVal,
+    deserialize::DeserializeValue,
+    frame::response::result::{ColumnType, CqlValue},
+    serialize::{
+        value::SerializeValue,
+        writers::{CellWriter, WrittenCellProof},
+        SerializationError,
+    },
+};
+
+use crate::{abs::PeekInterface, StrongSecret};
+
+impl<T> SerializeValue for StrongSecret<T>
+where
+    T: SerializeValue + zeroize::Zeroize + Clone,
+{
+    fn serialize<'b>(
+        &self,
+        typ: &ColumnType,
+        writer: CellWriter<'b>,
+    ) -> Result<WrittenCellProof<'b>, SerializationError> {
+        self.peek().serialize(typ, writer)
+    }
+}
+
+impl<'frame, T> DeserializeValue<'frame> for StrongSecret<T>
+where
+    T: DeserializeValue<'frame> + zeroize::Zeroize + Clone,
+{
+    fn type_check(_typ: &ColumnType) -> Result<(), scylla::deserialize::TypeCheckError> {
+        Ok(())
```

Comment:
Shouldn't we be calling `T::type_check(typ)` here?

Reply:
Missed it, fixed now